sdk/core/azure-core/inc/azure/core/http/body_stream.hpp

```diff
@@ -68,7 +68,10 @@ namespace Azure { namespace Core { namespace Http {
      */
     virtual void Rewind()
     {
-      throw "Not Implemented"; // TODO: Replace with best practice as defined by guideline
+      // Exception is not meant to be caught. This is rather an indication of a bad program that
+      // needs to be updated. This is the reason of using `throw` alone.
+      throw "The upload BodyStream doesn't support Rewind which is required to guarantee fault "
+            "tolerance when retrying the operation.";
```

Comment:
What is the exception type?

Reply:
`throw` alone is used for 2  cases:
1. re-throw an exception. When it is used inside a try/catch
2. terminate/abort the application.

https://stackoverflow.com/questions/4931867/throw-without-arguments-for-failure-signalling

Reply:
> terminate/abort the application.

In that case, should this be more like a precondition rather than a `throw`?

Reply:
It could be. What do we use for that: assert?